csrc/host_ir/container.h

```diff
@@ -46,6 +46,8 @@ class HostIrContainer final : public Fusion {
 
   void pushBackTopLevelExprs(Expr* expr);
 
+  void insertDeallocations();
```

Comment:
Consider insertExprAfter(int64_t index, Expr* expr) and making insertDeallocations non-class. 

A class method can access any class fields. While it's most flexible, access control can become a nightmare. 

I think a better design here is to add common access methods in HostIrContainer and move complicated transform logics (e.g. many host IR optimization passes we plan to write in the future) outside HostIrContainer.